linalg/ode.hpp

```diff
@@ -431,6 +431,48 @@ class SDIRK33Solver : public ODESolver
 };
 
 
+/** Two stage, explicit singly diagonal implicit Runge-Kutta (ESDIRK) method
+    of order 2. A-stable. */
+class TrapezoidalRuleSolver : public ODESolver
+{
+protected:
+   Vector k, y;
+
+public:
+   virtual void Init(TimeDependentOperator &_f);
+
+   virtual void Step(Vector &x, double &t, double &dt);
+};
+
+
+/** Three stage, explicit singly diagonal implicit Runge-Kutta (ESDIRK) method
+    of order 2. L-stable. */
+class ESDIRK32Solver : public ODESolver
```

Comment:
Just curious: how is a method "explicit" and "implicit" at the same time? 😄

Reply:
It's a mutant hybrid method! Probably there should be an arrow, E->SDIRK schemes, one explicit stage followed by a set of SDIRK stages